### ISSUE TITLE
Spring model generator - allOf does not work as expected

### ISSUE THREAD

**Issue by mresetar** [swagger-api/swagger-codegen#4467]
I'm trying to generate JSON model using the following spec:

```xml
                         <configuration>
                            <inputSpec>src/main/resources/swagger/my.yaml</inputSpec>
                            <language>spring</language>
                            <configOptions>
                                <sourceFolder>.</sourceFolder>
                                <dateLibrary>java8</dateLibrary>
                                <modelPackage>my.api.model</modelPackage>
                                <apiPackage>my.api.handler</apiPackage>
                                <!--<library>spring-mvc</library>-->
                            </configOptions>
                            <environmentVariables>
                                <models></models>
                                <DmodelDocs>false</DmodelDocs>
                            </environmentVariables>
                            <output>src/gen/java/main</output>
                        </configuration>
```
In Yaml I have taken an example from:
http://swagger.io/specification/#schemaObject

which is:

```yaml
     ErrorModel:
    type: object
    required:
    - message
    - code
    properties:
      message:
        type: string
      code:
        type: integer
        minimum: 100
        maximum: 600
        
  ExtendedErrorModel:
    allOf:
    - $ref: '#/definitions/ErrorModel'
    - type: object
      required:
      - rootCause
      properties:
        rootCause:
          type: string
```

I expect that generated Java model will reflect YAML but it does not. 
There are ErrorModel and ExtendedErrorModeljava classes but ExtendedErrorModel only has its own property ie "rootCause". 

I expect that ExtendedErrorModel extends ErrorModel or (maybe better) incorporates all attributes of ErrorModel. 

Model is defined as part of "definitions" of a Yaml (not inlined). 

I'm using 2.2.1 version of a plugin. 
Model is generated issuing mvn compile command. 

Miro

**Comment by shiqicao** [comment]
you need to add "discriminator" in ErrorModel

**Comment by chosroes** [comment]
@mresetar Did you solved this ? I have the same problem. With 2.1.6 the generated submodel extended the parent model correctly, but not anymore with 2.2.1

**Comment by mresetar** [comment]
At first, thank you @shiqicao for pointing me in right direction. Indeed, adding discriminator solves my issue. I'm still using 2.2.1 version.  @chosroes Take another look into your model as it works for me.

**Comment by chosroes** [comment]
@mresetar Not adapted for me. I don't want polymorphism. Just inheritance.

**Comment by wing328** [comment]
@chosroes please open an issue (if there isn't one opened yet) with details on what you want and ideally an example showing the difference between the existing output and the desired output.

**Comment by chosroes** [comment]
@wing328 I don't know if I should, since it works with http://editor.swagger.io/#/ which seems to use a more recent version of the code. Am I wrong ?

**Comment by wing328** [comment]
@mresetar I thought you didn't want polymorphism, just inheritance but that's not the current output. Sounds like it's not the case based on your reply (which means I misunderstood your reply)

**Comment by mresetar** [comment]
Hi, I wanted to (idealy) have polymorphism. @wing328 don't understand what is the the difference between inferitance and polymorphism for you. I agree with http://stackoverflow.com/questions/6308178/what-is-the-main-difference-between-inheritance-and-polymorphism  answer (Inheritance: This is one way of achieving both Polymorphism and code reuse at the same time.). So in this case to me is the same thing. 

With discriminator I have achieved inheritance like shown in diagram below:
[Diagram](http://2.1m.yt/nyjacUM.png)

YAML that triggers such class creation is as follows:

```
  ResponseGeneric:
      type: object
      discriminator: responseType
      required:
      - header
      - responseType
      properties:
        header:
          $ref: "#/definitions/Header"
        responseType:
          type: string
          enum: ["OK", "Error", "Warning"] 
        errors:
          type: array
          items:
            $ref: "#/definitions/Error"
        warnings:
          type: array
          items:
            $ref: "#/definitions/Warning"

  ErrorResponse:
    allOf:
    - $ref: "#/definitions/ResponseGeneric"
    - type: object

```
Basically ErrorResponse inherits ResponseGeneric and doesn't bring any new fields.

I hope this helps. Also read Swagger documentation for discrimator field.

**Comment by wing328** [comment]
I've to admit I've not read the whole discussion before my first reply. I found that this issue is closed and saw @chosroes reply, and that's why I suggest him to open a new issue instead if @chosroes is looking for something else.
